tests/Feature/Http/Controllers/EventControllerTest.php

```diff
@@ -35,15 +37,35 @@ public function setUp(): void
      */
     public function index_displays_view()
     {
-        $events = Event::factory()
-            ->count(3)
-            ->create();
-
-        $response = $this->get(route('event.index'));
+        $characterFactory = Character::factory()->has(User::factory());
+        $sessionFactory = Session::factory()->has($characterFactory);
+        $event = Event::factory()->has($sessionFactory)->create();
+        // Create events that _shouldn't_ be returned
+        Event::factory(3)->has(Session::factory(5))->create();
+        $user = $event->sessions[0]->characters[0]->user;
+
+        $response = $this->actingAs($user)->get(route('event.index', [
+            'search' => $event->title,
+            'registered_only' => true,
+            'registered_user' => $user->id,
```

Comment:
what about a test for if `registered_user` is null? i suspect it likely will be null since 99% of the time we're filtering this list for the logged in user, which `whereRegistered()` does automatically